### ISSUE TITLE
Support Python 3

### ISSUE THREAD

**Issue by mruffalo** [brentp/pyfasta#9]
It doesn't look there's much to do to allow pyfasta to run on Python 3. There are a few places in which bytes strings are used as text, which is fine in Python 2 since FASTA files are all ASCII.

To continue supporting Python 2.5 (which doesn't allow adding a 'b' before byte string literals), it may be necessary to open files in text mode instead of binary mode such as in lines 47 and 58 of records.py. This should have little to no effect in Python 2 except for newline conversion on Windows systems, and the "universal newlines" ( http://docs.python.org/3/glossary.html#term-universal-newlines ) mode is probably the right way to do this anyway. This will cause Python 3 to decode the file's bytes into Unicode characters, but the default encoding of UTF-8 will be fine since non-ASCII bytes are extremely rare in FASTA files.

**Comment by brentp** [comment]
I am fine to drop python 2.5 support if you want to make a pull request.

**Comment by mruffalo** [comment]
It turns out that the NumPy-backed records are what gave me the most trouble, and that being able to prefix a byte literal with "b" isn't really an issue at all.

What _would_ be convenient though, is changing the Fasta class to extend collections.abc.Mapping ( http://docs.python.org/3/library/collections.abc.html#collections.abc.Mapping ), which was added in 2.6. Subclassing dict has some side effects in Python 3, notably that the items() method no longer returns a list of items but instead a "view" of the underlying data. I had to override items() to avoid looking at the internal dict data that's provided by the superclass.

At the moment all tests pass except for a very strange assertion failure in test_array_copy -- the last one, that checks against 'NNNN'. The data matches when I test it myself in the interactive interpreter, though.

I'll prepare one pull request that supports Python 2.5 through 3.3 inclusive, and another that drops support for 2.5. In addition to being able to use the Mapping abstract base class, there are a lot of things that can be done in 2.6 to help the automated 2to3 conversion produce better code.
